### PR TITLE
Add front shortlist

### PR DESCRIPTION
**Pull request by l-vincent-l**
Partial solution for #22 

- [x] There's a list of available, usable data
- [x] I can know the coverage of the data
- [x] I can know the LICENSE
- [x] I can know the format
- [x] I can know how to contact the AO
- [x] Add internationalization
- [x] Add download button
- [x] Add tests

Preview:

image

### CODE REVIEW

File: client/javascripts/components/shortlist/shortlist.tag [hunk 1]
```diff
@@ -0,0 +1,60 @@
+<shortlist>
+<div>
```
Comment (maukoquiroga) [inline client/javascripts/components/shortlist/shortlist.tag]: Indentation

File: client/stylesheets/components/_shortlist.scss [hunk 2]
```diff
@@ -0,0 +1,79 @@
+shortlist {
+ $base-spacing: 1.5em !default;
+ $dark-gray: #333 !default;
+ $light-gray: #777 !default;
+ $dark-blue: #244e87 !default;
+ $base-font-color: $dark-gray !default;
+ $shortlist-gutter: 1.4em;
+ $shortlist-image-padding: 0.7em;
```
Comment (maukoquiroga) [inline client/stylesheets/components/_shortlist.scss]: Prefer `.7em`

File: client/stylesheets/components/_shortlist.scss [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+ $shortlist-image-width: 4em;
+ $shortlist-color: $base-font-color;
+ $shortlist-image-vert-alignment: top;
+ > div {
+   padding-top: 100px;
+ }
+
+.shortlist {
+
```
Comment (maukoquiroga) [inline client/stylesheets/components/_shortlist.scss]: Unnecessary space.

File: lib/transport_web/templates/page/shortlist.html.eex [hunk 4]
```diff
@@ -0,0 +1,7 @@
+<h1><%= gettext "availabledatasets" %></h1>
```
Comment (maukoquiroga) [inline lib/transport_web/templates/page/shortlist.html.eex]: I'd encode them: `shortlist_title`, `shortlist_download`, etc.

File: client/javascripts/components/shortlist/shortlist.tag [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+          </div>
+        </div>
+      </div>
+    </div>
+  </div>
+</div>
+
+
+<script type="es6">
+console.log(this.opts)
```
Comment (maukoquiroga) [inline client/javascripts/components/shortlist/shortlist.tag]: Leftover

File: client/javascripts/components/shortlist/shortlist.tag [hunk 6]
```diff
@@ -0,0 +1,64 @@
+<shortlist>
```
Comment (maukoquiroga) [inline client/javascripts/components/shortlist/shortlist.tag]: Following the other components, and site wide style, indentation should be of 4 spaces.

File: client/stylesheets/_variables.scss [hunk 7]
```diff
@@ -1,5 +1,8 @@
 // Colours
 $light-grey: hsl(0, 0%, 93%);
+$black: #000;
+$dark-grey: #333 !default;
```
Comment (maukoquiroga) [inline client/stylesheets/_variables.scss]: Should add default to all ?

File: client/stylesheets/components/_shortlist.scss [hunk 8]
```diff
@@ -0,0 +1,95 @@
+shortlist {
+  $base-spacing: 1.5em !default;
+  $shortlist-gutter: 1.4em;
+  $shortlist-image-padding: .7em;
+  $shortlist-image-width: 4em;
+  $shortlist-image-vert-alignment: top;
+
+  $shortlist-layout-grid: (
```
Comment (maukoquiroga) [inline client/stylesheets/components/_shortlist.scss]: Can reuse mixins.

File: client/stylesheets/components/_shortlist.scss [hunk 9]
```diff
@@ -0,0 +1,95 @@
+shortlist {
+  $base-spacing: 1.5em !default;
```
Comment (maukoquiroga) [inline client/stylesheets/components/_shortlist.scss]: Here `!default` make sense, as `$base-spacing` could already be defined and shared among components.

File: client/javascripts/components/shortlist/shortlist.tag (hunk 6 above)
Comment (l-vincent-l) [reply client/javascripts/components/shortlist/shortlist.tag]: done.

File: client/javascripts/components/shortlist/shortlist.tag [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+                        </div>
+                    </div>
+                </div>
+            </div>
+        </div>
+    </div>
+
+
+<script type="es6">
+console.log(this.opts)
```
Comment (maukoquiroga) [inline client/javascripts/components/shortlist/shortlist.tag]: Leftover

File: client/javascripts/components/shortlist/shortlist.tag [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+this.shortlist = []
+
+this.fetch_shortlist = () => {
```
Comment (maukoquiroga) [inline client/javascripts/components/shortlist/shortlist.tag]: Indentation should be applied here as well

File: client/stylesheets/components/_shortlist.scss [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+    }
+
+    .shortlist-description {
+      h1 {
+        font-size: 1em;
+        margin: 0 0 .5em;
+        text-align: left;
+      }
+
+      markdown {
```
Comment (maukoquiroga) [inline client/stylesheets/components/_shortlist.scss]: It is not the same as ```markdown > div {``` ?

File: test/transport_web/solutions/reuse_data_test.exs [hunk 13]
```diff
@@ -12,7 +12,7 @@ defmodule TransportWeb.ReuseDataTest do
   hound_session()
 
   @tag :integration
-  test "I can click somewhere to reuse transport data" do
+  test "I can click on the map to reuse transport data" do
```
Comment (maukoquiroga) [inline test/transport_web/solutions/reuse_data_test.exs]: This test should fail now.

Comment (l-vincent-l) [reply test/transport_web/solutions/reuse_data_test.exs]: Nop, we still need a page per organization to change this test

**Comment by maukoquiroga** [comment]
Map buttons as in master:

capture d ecran 2017-10-03 a 14 07 13

And this branch:

capture d ecran 2017-10-03 a 14 08 28

File: test/transport_web/solutions/reuse_data_test.exs (hunk 13 above)
Comment (maukoquiroga) [reply test/transport_web/solutions/reuse_data_test.exs]: Why ? Wouldn't it be enough to add a link to the .zip file ?

Comment (l-vincent-l) [reply test/transport_web/solutions/reuse_data_test.exs]: I'm not a fan, if you're using the site for the first time you'll want info about the licence for example.

**Comment by l-vincent-l** [comment]
This is fixed

File: test/transport_web/solutions/reuse_data_test.exs (hunk 13 above)
Comment (maukoquiroga) [reply test/transport_web/solutions/reuse_data_test.exs]: OK

**Pull request merged**